cli/tools/upgrade.rs

```diff
@@ -72,20 +75,57 @@ pub fn check_for_upgrades(cache_dir: PathBuf) -> Option<String> {
       };
 
       let contents = CheckVersionFile {
+        last_prompt: chrono::Utc::now(),
         last_checked: chrono::Utc::now(),
         latest_version,
       }
       .serialize();
-      let _ = std::fs::write(cache_dir.join(UPGRADE_CHECK_FILE_NAME), contents);
+      let _ =
+        std::fs::write(cache_dir1.join(UPGRADE_CHECK_FILE_NAME), contents);
```

Comment:
Might be nice to have a `check_file.save(cache_dir);` method then re-use that here and below.